src/staticanalysis/bot/static_analysis_bot/workflow.py

```diff
@@ -237,6 +239,14 @@ def run(self, revision):
             self.index(revision, state='done', issues=0)
             return
 
+        # Publish patches on Taskcluster
+        # or write locally for local development
+        for patch in revision.improvement_patches:
+            if settings.taskcluster.local:
```

Comment:
It's a bit ugly to have different behavior here, I wonder if we can not just call publish but then avoid publishing when we're local

Reply:
I disagree ! I find it useful to have local output during dev (and we can't publish at all on Taskcluster artifacts from our computers)

Reply:
Sorry, I didn't mean to completely remove it, I meant we could move this check in publish, so we have the is-local check only at the end of the process.